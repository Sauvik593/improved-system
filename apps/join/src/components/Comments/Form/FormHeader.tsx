import { ReplyData } from '@lib/api/strapi/addComment';
import { useTranslation } from 'next-i18next';

interface Props {
  parent?: ReplyData;
}

const headerClassName = 'text-h-4-sm md:text-h-4 text-tile-100 font-bold mb-3 comment-header';

export const AddCommentHeader = () => {
  const { t } = useTranslation('common');
  return (
    <p className={headerClassName} id="add-voice">
      {t('content.comments.add')}
    </p>
  );
};

export const ReplyCommentHeader = ({ parent }: Props) => {
  const { t } = useTranslation('common');
  return (
    <p className={headerClassName} id={`reply_to_${parent?.parentId}`}>
      {t('content.comments.reply', { user: parent?.user })}
    </p>
  );
};
