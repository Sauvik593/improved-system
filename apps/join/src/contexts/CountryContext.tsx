import { createContext } from 'react';
import type { Country } from '@lib/types';

export interface CountryContextParams {
  countries: Country[];
  country: Country | null;
}

export const CountryContext = createContext<CountryContextParams>({
  countries: [],
  country: null,
});
