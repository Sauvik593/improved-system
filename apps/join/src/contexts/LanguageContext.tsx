import { createContext } from 'react';

export interface LanguageContextParams {
  locale: string;
  locales: string[];
}

export const LanguageContext = createContext<LanguageContextParams>({
  locale: '',
  locales: [],
});
