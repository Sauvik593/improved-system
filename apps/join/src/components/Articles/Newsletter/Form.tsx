import { useContext } from 'react';
import { useTranslation } from 'next-i18next';
import { startCase } from 'lodash';
import Link from 'next/link';
import type { UseFormRegister, FieldErrors } from 'react-hook-form';

import { BaseInput } from '@components/common/FormInputs/BaseInput';
import { HiddenInput } from '@components/common/FormInputs/HiddenInput';
import { LoadingButton } from '@components/common/LoadingButton';

import { LanguageContext } from '@contexts/LanguageContext';
import { CountryContext } from '@contexts/CountryContext';

import type { NewsletterFormProps } from '@components/Subscribe/Form/submitForm';
import { createDocsPath } from '@helpers/links';

interface Props {
  handleSubmit: () => void;
  register: UseFormRegister<NewsletterFormProps>;
  errors: FieldErrors<NewsletterFormProps>;
  submitting: boolean;
}

export const Form = ({ handleSubmit, register, errors, submitting }: Props) => {
  const { locale } = useContext(LanguageContext);
  const { country } = useContext(CountryContext);
  const docsPath = createDocsPath(locale, ['privacy']);
  const { t } = useTranslation('common');
  return (
    <section>
      <header>
        <h3 className="text-h-4 mb-3 font-semibold">{t('insights_newsletter.title')}</h3>
        <p className="mb-3">{t('insights_newsletter.description')}</p>
      </header>
      <form onSubmit={handleSubmit}>
        <BaseInput
          name="email"
          type="email"
          labelText="text-white"
          label={t('newsletter.email.label')}
          error={errors.email}
          register={register}
        />
        <HiddenInput name="hsLanguage" register={register} value={locale} />
        <HiddenInput
          name="defaultNationContact"
          register={register}
          value={startCase(country?.translation_key as string)}
        />
        <HiddenInput name="receiveMarketAnalysis" register={register} value={'true'} />
        <div className="mt-6">
          <LoadingButton
            color="sky"
            size="normal"
            loading={submitting}
            label={t('ui.buttons.subscribe')}
          />
        </div>
      </form>
      <footer className="text-p-3 mt-2 px-2 text-center font-semibold">
        {t('insights_newsletter.footer.privacy')}
        <Link href={docsPath}>
          <a className="ml-2 text-sky-100 hover:underline">
            {t('insights_newsletter.footer.policy')}
          </a>
        </Link>
      </footer>
    </section>
  );
};
