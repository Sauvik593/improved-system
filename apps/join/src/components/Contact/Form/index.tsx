import { useContext } from 'react';
import { startCase } from 'lodash';
import { Controller, useForm } from 'react-hook-form';
import { useTranslation } from 'next-i18next';

import { createContactFormSchema } from './createContactFormSchema';
import { yupResolver } from '@hookform/resolvers/yup';

import { BaseInput } from '@components/common/FormInputs/BaseInput';
import { HiddenInput } from '@components/common/FormInputs/HiddenInput';
import { SelectInput } from '@components/common/FormInputs/SelectInput';
import { ContactFormProps } from './submitForm';
import { submitForm } from './submitForm';
import { SuccessModal } from '../SuccessModal';
import { useHubspotForm } from '@hooks/useHubspotForm';
import { LanguageContext } from '@contexts/LanguageContext';

import { Radio } from '@kyero/ui';
import { ErrorModal } from '@components/common/ErrorModal';
import { LoadingButton } from '@components/common/LoadingButton';
import { CountryContext } from '@contexts/CountryContext';
import { mapCitiesToOptions, PROPERTIES_NUMBER, TIME_TO_CALL, GET_IN_TOUCH } from './formOptions';

export const ContactForm = () => {
  const { locale } = useContext(LanguageContext);
  const { country } = useContext(CountryContext);
  const CITY_OPTIONS = mapCitiesToOptions(country?.translation_key as string);

  const { t } = useTranslation('common');
  const BASED_OPTIONS = [
    { value: 'Advertiser', label: t('contact.form.user_type.options.new') },
    { value: 'Estate Agent', label: t('contact.form.user_type.options.current_agent') },
    { value: 'Developer', label: t('contact.form.user_type.options.developer') },
    { value: 'Buyer', label: t('contact.form.user_type.options.buyer') },
    { value: 'Other', label: t('contact.form.user_type.options.other') },
  ];

  const {
    register,
    control,
    handleSubmit,
    getValues,
    reset,
    formState: { errors, isSubmitting },
  } = useForm<ContactFormProps>({
    resolver: yupResolver(createContactFormSchema(t)),
  });
  const { error, setFormState, submit, success } = useHubspotForm({
    formService: () => submitForm(getValues()),
  });

  const closeModal = () => {
    setFormState('form');
    reset();
  };
  return (
    <section data-testid="contact-wrapper">
      <form onSubmit={handleSubmit(submit)} className="text-tile-100" data-testid="form">
        <h3 className="text-h-3-sm lg:text-h-3 text-tile-100 my-2 font-bold">
          {t('contact.form.header')}
        </h3>
        <p className="mb-4">{t('contact.form.contact_information')}</p>
        <div className="mb-4 grid grid-rows-none gap-y-4 lg:grid-cols-2 lg:gap-x-6">
          <BaseInput
            name={'firstName'}
            label={t('contact.form.first_name.label')}
            register={register}
            error={errors.firstName}
            testId="first-name"
          />

          <BaseInput
            name={'lastName'}
            label={t('contact.form.last_name.label')}
            register={register}
            error={errors.lastName}
            testId="last-name"
          />

          <BaseInput
            name={'email'}
            label={t('contact.form.work_email.label')}
            register={register}
            type="email"
            error={errors.email}
            testId="email"
          />

          <BaseInput
            name={'phone'}
            label={t('contact.form.phone_number.label')}
            register={register}
            defaultValue={null}
            error={errors.phone}
            testId="phone-number"
          />
        </div>
        <div>
          <p className="mb-2">{t('contact.form.business_information')}</p>
          <div className="mb-4 gap-y-4 lg:grid lg:grid-cols-2 lg:items-end lg:gap-x-6">
            <BaseInput
              className="my-2"
              name={'company'}
              label={t('contact.form.business_name.label')}
              register={register}
              error={errors.company}
              testId="company-name"
            />
            <div className="py-2">
              <Controller
                name="userType"
                control={control}
                render={({ field: { onChange, name, value } }) => (
                  <SelectInput
                    value={value}
                    name={name}
                    onChange={onChange}
                    options={BASED_OPTIONS}
                    placeholder={t('contact.form.select.placeholder')}
                    label={t('contact.form.user_type.label')}
                    testId="user-type"
                    error={errors.userType}
                  />
                )}
              />
            </div>
          </div>
          <div className="my-6 space-y-4 lg:grid lg:grid-cols-2 lg:items-end lg:gap-x-6 lg:space-y-0">
            <Controller
              name="agentMainProvinceContact"
              control={control}
              render={({ field: { onChange, name, value } }) => (
                <SelectInput
                  value={value}
                  name={name}
                  onChange={onChange}
                  options={CITY_OPTIONS}
                  placeholder={t('contact.form.select.placeholder')}
                  label={t('contact.form.agent_province.label')}
                />
              )}
            />

            <Controller
              name="numberOfPropertiesEnquire"
              control={control}
              render={({ field: { onChange, name, value } }) => (
                <SelectInput
                  value={value}
                  name={name}
                  onChange={onChange}
                  options={PROPERTIES_NUMBER}
                  placeholder={t('contact.form.select.placeholder')}
                  label={t('contact.form.properties_number.label')}
                />
              )}
            />
          </div>
        </div>

        <div>
          <div className="mb-5 items-end lg:grid lg:grid-cols-2 lg:gap-x-6">
            <div>
              <p className="text-p-2 md:text-h-6 mb-4 font-bold">
                {t('contact.form.arrange_callback')}
              </p>
              <div className="space-y-4 lg:flex lg:items-center lg:space-y-0 lg:space-x-4">
                <Controller
                  name="timeOfDay"
                  control={control}
                  render={({ field: { onChange, name } }) => (
                    <>
                      {TIME_TO_CALL.map((option) => (
                        <Radio
                          onChange={onChange}
                          key={option.label}
                          defaultChecked={false}
                          label={t(option.label)}
                          name={name}
                          value={option.value}
                        />
                      ))}
                    </>
                  )}
                />
              </div>
            </div>
            <div className="mt-5 lg:mt-0">
              <p className="text-p-2 md:text-h-6 mb-4 font-bold">
                {t('contact.form.get_in_touch.label')}
              </p>
              <div className="space-y-4 lg:flex lg:items-center lg:space-y-0 lg:space-x-4">
                <Controller
                  name="preferredFormOfContact"
                  control={control}
                  render={({ field: { onChange, name } }) => (
                    <>
                      {GET_IN_TOUCH.map((option) => (
                        <Radio
                          onChange={onChange}
                          key={option.label}
                          defaultChecked={false}
                          label={t(option.label)}
                          name={name}
                          value={option.value}
                        />
                      ))}
                    </>
                  )}
                />
              </div>
            </div>
          </div>
        </div>
        <HiddenInput
          name={'defaultNationContact'}
          register={register}
          value={startCase(country?.translation_key as string)}
        />
        <HiddenInput name={'hsLanguage'} register={register} value={locale} />
        <HiddenInput name={'receiveMarketAnalysis'} register={register} value={'true'} />
        <footer className="text-h-5-sm lg:text-h-5 mt-3 w-full lg:w-1/4">
          <LoadingButton loading={isSubmitting} label={t('contact.form.submit')} />
        </footer>
      </form>
      <SuccessModal active={success} onClose={closeModal} email={getValues('email')} />
      <ErrorModal active={error} onClose={() => setFormState('form')} />
    </section>
  );
};
