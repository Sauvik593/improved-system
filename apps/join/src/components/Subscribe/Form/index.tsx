import { useContext } from 'react';
import type { UseFormRegister, FieldErrors } from 'react-hook-form';

import { useTranslation } from 'next-i18next';
import { startCase } from 'lodash';

import { LoadingButton } from '@components/common/LoadingButton';
import { BaseInput } from '@components/common/FormInputs/BaseInput';
import { HiddenInput } from '@components/common/FormInputs/HiddenInput';
import { LanguageContext } from '@contexts/LanguageContext';
import { CountryContext } from '@contexts/CountryContext';
import { NewsletterFormProps } from './submitForm';
import { NewsletterFooter } from '../NewsletterFooter';

interface Props {
  handleSubmit: () => void;
  register: UseFormRegister<NewsletterFormProps>;
  errors: FieldErrors<NewsletterFormProps>;
  submitting: boolean;
}

export const Form = ({ handleSubmit, register, errors, submitting }: Props) => {
  const { t } = useTranslation('common');
  const { locale } = useContext(LanguageContext);
  const { country } = useContext(CountryContext);
  return (
    <div data-testid="form-wrapper" className="lg:max-w-3xl">
      <h3 className="text-h-4-sm lg:text-h-4 mx-auto mb-4 text-center font-bold">
        {t('newsletter.title')}
      </h3>
      <form onSubmit={handleSubmit} data-testid="form">
        <div className="mx-auto mb-5 w-3/4 lg:grid lg:grid-cols-3 lg:gap-2">
          <div className="w-full lg:col-span-2 lg:mr-4">
            <BaseInput
              name="email"
              type="email"
              placeholder={t('newsletter.email.placeholder')}
              register={register}
              error={errors.email}
              testId="email"
              showTooltip
            />
          </div>
          <HiddenInput name="hsLanguage" register={register} value={locale} />
          <HiddenInput
            name="defaultNationContact"
            register={register}
            value={startCase(country?.translation_key as string)}
          />
          <HiddenInput name="receiveMarketAnalysis" register={register} value={'true'} />
          <div className="text-h-5-sm lg:text-h-5 mt-4 lg:mt-0 lg:grid-cols-1">
            <LoadingButton
              color="sky"
              size="normal"
              loading={submitting}
              label={t('ui.buttons.subscribe')}
            />
          </div>
        </div>
        <NewsletterFooter />
      </form>
    </div>
  );
};
