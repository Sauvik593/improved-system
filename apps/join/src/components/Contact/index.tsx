import { useTranslation } from 'next-i18next';

import { Hero } from '@components/common/Hero';
import { Testimonials } from '../Testimonials';
import { TrustedAgents } from '../TrustedAgents';
import { ContactForm } from './Form';
import { FormFooter } from './Form/Footer';
import type { AgentsSeed } from '@components/TrustedAgents/AgentsSeed';
import type { TestimonialsSeed } from '@components/Testimonials/TestimonialsSeed';

interface Props {
  agentsSeed: AgentsSeed;
  testimonialsSeed: TestimonialsSeed;
}

export const ContactDetails = ({ agentsSeed, testimonialsSeed }: Props) => {
  const { t } = useTranslation('common');
  return (
    <>
      <section className="bg-sierra-night-5 relative">
        <Hero
          title={t('contact.title')}
          description={t('contact.description')}
          type="left"
          paddingClassName="pt-16 pb-20 md:pb-48"
        />
        <div className="container relative mx-auto -mt-20 md:-mt-44">
          <div className="max-w-3xl rounded-xl bg-white shadow-sm">
            <div className="py-4 px-6">
              <div id="hubspot-form-wrapper">
                <ContactForm />
              </div>
            </div>
            <FormFooter />
          </div>
          <Testimonials testimonialsSeed={testimonialsSeed} />
        </div>
        <div className="pb-12 md:pb-24">
          <TrustedAgents agentsSeed={agentsSeed} />
        </div>
      </section>
    </>
  );
};
