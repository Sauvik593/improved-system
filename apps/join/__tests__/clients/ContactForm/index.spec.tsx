import {
  render,
  screen,
  waitFor,
  getByTestId,
  fireEvent,
  act,
  RenderResult,
} from '@testing-library/react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';

import { ContactForm } from '@components/Contact/Form';

import { setupIntersectionObserverMock } from '../../intersectionObserverMock';
import { CountryContext } from '@contexts/CountryContext';
import { LanguageContext } from '@contexts/LanguageContext';
import { countrySpain, mockedCountries } from '../../countriesMock';

interface FormValues {
  email: string;
  phoneNumber: string;
  firstName: string;
  lastName: string;
  businessName: string;
  userType: string;
}

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: string) => key }),
}));

const countries = mockedCountries;

const server = setupServer(
  rest.post('/new-join/api/hubspot', (_req, res, ctx) => res(ctx.status(200))),
);

const submitForm = async (values: FormValues): Promise<void> => {
  const contactWrapper = screen.getByTestId('contact-wrapper');
  const emailField = getByTestId(contactWrapper, 'email') as HTMLInputElement;
  const phoneNumberField = getByTestId(contactWrapper, 'phone-number') as HTMLInputElement;
  const firstNameField = getByTestId(contactWrapper, 'first-name') as HTMLInputElement;
  const lastNameField = getByTestId(contactWrapper, 'last-name') as HTMLInputElement;
  const businessNameField = getByTestId(contactWrapper, 'company-name') as HTMLInputElement;
  const userTypeField = getByTestId(contactWrapper, 'user-type') as HTMLSelectElement;
  const form = getByTestId(contactWrapper, 'form') as HTMLFormElement;

  await act(async () => {
    await waitFor(async () => {
      fireEvent.change(emailField, { target: { value: values.email } });
      fireEvent.change(phoneNumberField, { target: { value: values.phoneNumber } });
      fireEvent.change(firstNameField, { target: { value: values.firstName } });
      fireEvent.change(lastNameField, { target: { value: values.lastName } });
      fireEvent.change(businessNameField, { target: { value: values.businessName } });
      fireEvent.change(userTypeField, { target: { value: values.userType } });
      fireEvent.submit(form);
    });
  });
};

describe('ContactForm', () => {
  let renderResult: RenderResult;

  beforeEach(() => {
    renderResult = render(
      <CountryContext.Provider value={{ country: countrySpain, countries }}>
        <LanguageContext.Provider value={{ locale: 'en', locales: ['en', 'es', 'pt', 'fr', 'it'] }}>
          <ContactForm />
        </LanguageContext.Provider>
      </CountryContext.Provider>,
    );
    setupIntersectionObserverMock();
  });

  beforeAll(() => {
    server.listen();
  });

  afterEach(() => {
    server.resetHandlers();
  });

  afterAll(() => server.close());

  describe('When initially rendered', () => {
    it('displays the form view', () => {
      render(
        <CountryContext.Provider value={{ country: countrySpain, countries }}>
          <ContactForm />
        </CountryContext.Provider>,
      );
      const successModal = screen.queryByTestId('success-modal');
      const errorModal = screen.queryByTestId('error-modal');

      expect(successModal).not.toBeInTheDocument();
      expect(errorModal).not.toBeInTheDocument();
    });
  });

  describe('When the form is submitted', () => {
    it('displays the form success', async () => {
      await submitForm({
        email: 'test@test.com',
        phoneNumber: '1234567890',
        firstName: 'John',
        lastName: 'Doe',
        businessName: 'ACME INC',
        userType: 'Developer',
      });

      await waitFor(() => {
        expect(screen.queryByTestId('success-modal')).toBeInTheDocument();
        expect(screen.queryByTestId('error-modal')).not.toBeInTheDocument();
      });
    });
  });

  describe('When the form is submitted but request fails', () => {
    it('displays the error modal', async () => {
      server.use(rest.post('/new-join/api/hubspot', (req, res, ctx) => res(ctx.status(500))));

      await submitForm({
        email: 'test@test.com',
        phoneNumber: '1234567890',
        firstName: 'John',
        lastName: 'Doe',
        businessName: 'ACME INC',
        userType: 'Developer',
      });

      await waitFor(() => {
        expect(screen.queryByTestId('error-modal')).toBeInTheDocument();
        expect(screen.queryByTestId('success-modal')).not.toBeInTheDocument();
      });
    });
  });
  describe('When form is filled', () => {
    it('shows validation errors', async () => {
      const { asFragment } = renderResult;

      await submitForm({
        email: 'not_valid_email',
        phoneNumber: '123',
        firstName: '',
        lastName: '',
        businessName: '',
        userType: '',
      });

      await waitFor(() => {
        expect(screen.queryByTestId('first-name-error')).toBeInTheDocument();
        expect(screen.queryByTestId('last-name-error')).toBeInTheDocument();
        expect(screen.queryByTestId('email-error')).toBeInTheDocument();
        expect(screen.queryByTestId('phone-number-error')).toBeInTheDocument();
        expect(screen.queryByTestId('company-name-error')).toBeInTheDocument();
        expect(screen.queryByTestId('user-type-error')).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
      });
    });
  });
});
