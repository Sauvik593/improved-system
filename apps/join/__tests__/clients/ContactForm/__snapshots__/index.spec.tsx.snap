// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm When form is filled shows validation errors 1`] = `
<DocumentFragment>
  <section
    data-testid="contact-wrapper"
  >
    <form
      class="text-tile-100"
      data-testid="form"
    >
      <h3
        class="text-h-3-sm lg:text-h-3 text-tile-100 my-2 font-bold"
      >
        contact.form.header
      </h3>
      <p
        class="mb-4"
      >
        contact.form.contact_information
      </p>
      <div
        class="mb-4 grid grid-rows-none gap-y-4 lg:grid-cols-2 lg:gap-x-6"
      >
        <div>
          <label
            class="text-p-2 md:text-h-6 font-bold text-tile-100"
            for="firstName"
          >
            contact.form.first_name.label
          </label>
          <input
            class="border-sierra-night-40 text-sierra-night-60 w-full rounded-md border p-2 pl-4 mt-2"
            data-testid="first-name"
            name="firstName"
            type="text"
          />
          <p
            class="px-2 text-terracotta-100 mt-2"
            data-testid="first-name-error"
          >
            form.validations.first_name
          </p>
        </div>
        <div>
          <label
            class="text-p-2 md:text-h-6 font-bold text-tile-100"
            for="lastName"
          >
            contact.form.last_name.label
          </label>
          <input
            class="border-sierra-night-40 text-sierra-night-60 w-full rounded-md border p-2 pl-4 mt-2"
            data-testid="last-name"
            name="lastName"
            type="text"
          />
          <p
            class="px-2 text-terracotta-100 mt-2"
            data-testid="last-name-error"
          >
            form.validations.last_name
          </p>
        </div>
        <div>
          <label
            class="text-p-2 md:text-h-6 font-bold text-tile-100"
            for="email"
          >
            contact.form.work_email.label
          </label>
          <input
            class="border-sierra-night-40 text-sierra-night-60 w-full rounded-md border p-2 pl-4 mt-2"
            data-testid="email"
            name="email"
            type="email"
          />
          <p
            class="px-2 text-terracotta-100 mt-2"
            data-testid="email-error"
          >
            form.validations.email.format
          </p>
        </div>
        <div>
          <label
            class="text-p-2 md:text-h-6 font-bold text-tile-100"
            for="phone"
          >
            contact.form.phone_number.label
          </label>
          <input
            class="border-sierra-night-40 text-sierra-night-60 w-full rounded-md border p-2 pl-4 mt-2"
            data-testid="phone-number"
            name="phone"
            type="text"
          />
          <p
            class="px-2 text-terracotta-100 mt-2"
            data-testid="phone-number-error"
          >
            form.validations.phone_number.format
          </p>
        </div>
      </div>
      <div>
        <p
          class="mb-2"
        >
          contact.form.business_information
        </p>
        <div
          class="mb-4 gap-y-4 lg:grid lg:grid-cols-2 lg:items-end lg:gap-x-6"
        >
          <div>
            <label
              class="text-p-2 md:text-h-6 font-bold text-tile-100"
              for="company"
            >
              contact.form.business_name.label
            </label>
            <input
              class="border-sierra-night-40 text-sierra-night-60 w-full rounded-md border p-2 pl-4 my-2 mt-2"
              data-testid="company-name"
              name="company"
              type="text"
            />
            <p
              class="px-2 text-terracotta-100 mt-2"
              data-testid="company-name-error"
            >
              form.validations.business_name
            </p>
          </div>
          <div
            class="py-2"
          >
            <div>
              <label
                class="text-p-2 md:text-h-6 font-bold text-tile-100"
                for="userType"
              >
                contact.form.user_type.label
              </label>
              <select
                class="border-sierra-night-40 text-sierra-night-60 w-full rounded-md border p-2 pl-4 mt-2"
                data-testid="user-type"
              >
                <option
                  disabled=""
                  selected=""
                  value=""
                >
                  contact.form.select.placeholder
                </option>
                <option
                  value="Advertiser"
                >
                  contact.form.user_type.options.new
                </option>
                <option
                  value="Estate Agent"
                >
                  contact.form.user_type.options.current_agent
                </option>
                <option
                  value="Developer"
                >
                  contact.form.user_type.options.developer
                </option>
                <option
                  value="Buyer"
                >
                  contact.form.user_type.options.buyer
                </option>
                <option
                  value="Other"
                >
                  contact.form.user_type.options.other
                </option>
              </select>
              <p
                class="text-terracotta-100 mt-2 px-2"
                data-testid="user-type-error"
              >
                contact.form.user_type.required
              </p>
            </div>
          </div>
        </div>
        <div
          class="my-6 space-y-4 lg:grid lg:grid-cols-2 lg:items-end lg:gap-x-6 lg:space-y-0"
        >
          <div>
            <label
              class="text-p-2 md:text-h-6 font-bold text-tile-100"
              for="agentMainProvinceContact"
            >
              contact.form.agent_province.label
            </label>
            <select
              class="border-sierra-night-40 text-sierra-night-60 w-full rounded-md border p-2 pl-4 mt-2"
            >
              <option
                disabled=""
                selected=""
                value=""
              >
                contact.form.select.placeholder
              </option>
              <option
                value="Alava"
              >
                Alava
              </option>
              <option
                value="Albacete"
              >
                Albacete
              </option>
              <option
                value="Alicante"
              >
                Alicante
              </option>
              <option
                value="Almeria"
              >
                Almeria
              </option>
              <option
                value="Asturias"
              >
                Asturias
              </option>
              <option
                value="Avila"
              >
                Avila
              </option>
              <option
                value="Badajoz"
              >
                Badajoz
              </option>
              <option
                value="Barcelona"
              >
                Barcelona
              </option>
              <option
                value="Burgos"
              >
                Burgos
              </option>
              <option
                value="Caceres"
              >
                Caceres
              </option>
              <option
                value="Cadiz"
              >
                Cadiz
              </option>
              <option
                value="Cantabria"
              >
                Cantabria
              </option>
              <option
                value="Castellon"
              >
                Castellon
              </option>
              <option
                value="Ciudad Real"
              >
                Ciudad Real
              </option>
              <option
                value="Cordoba"
              >
                Cordoba
              </option>
              <option
                value="Coruna"
              >
                Coruna
              </option>
              <option
                value="Cuenca"
              >
                Cuenca
              </option>
              <option
                value="El Hierro"
              >
                El Hierro
              </option>
              <option
                value="Formentera"
              >
                Formentera
              </option>
              <option
                value="Fuertaventura"
              >
                Fuertaventura
              </option>
              <option
                value="Gerona"
              >
                Gerona
              </option>
              <option
                value="Gran Canaria"
              >
                Gran Canaria
              </option>
              <option
                value="Granada"
              >
                Granada
              </option>
              <option
                value="Guadalajara"
              >
                Guadalajara
              </option>
              <option
                value="Guipuzcoa"
              >
                Guipuzcoa
              </option>
              <option
                value="Huelva"
              >
                Huelva
              </option>
              <option
                value="Huesca"
              >
                Huesca
              </option>
              <option
                value="Ibiza"
              >
                Ibiza
              </option>
              <option
                value="Jaen"
              >
                Jaen
              </option>
              <option
                value="La Palma"
              >
                La Palma
              </option>
              <option
                value="La Rioja"
              >
                La Rioja
              </option>
              <option
                value="Lanzarote"
              >
                Lanzarote
              </option>
              <option
                value="Leon"
              >
                Leon
              </option>
              <option
                value="Lerida"
              >
                Lerida
              </option>
              <option
                value="Lugo"
              >
                Lugo
              </option>
              <option
                value="Madrid"
              >
                Madrid
              </option>
              <option
                value="Malaga"
              >
                Malaga
              </option>
              <option
                value="Mallorca"
              >
                Mallorca
              </option>
              <option
                value="Menorca"
              >
                Menorca
              </option>
              <option
                value="Murcia"
              >
                Murcia
              </option>
              <option
                value="Navarra"
              >
                Navarra
              </option>
              <option
                value="Orense"
              >
                Orense
              </option>
              <option
                value="Palencia"
              >
                Palencia
              </option>
              <option
                value="Pontevedra"
              >
                Pontevedra
              </option>
              <option
                value="Salamanca"
              >
                Salamanca
              </option>
              <option
                value="Segovia"
              >
                Segovia
              </option>
              <option
                value="Sevilla"
              >
                Sevilla
              </option>
              <option
                value="Soria"
              >
                Soria
              </option>
              <option
                value="Tarragona"
              >
                Tarragona
              </option>
              <option
                value="Tenerife"
              >
                Tenerife
              </option>
              <option
                value="Teruel"
              >
                Teruel
              </option>
              <option
                value="Toledo"
              >
                Toledo
              </option>
              <option
                value="Valencia"
              >
                Valencia
              </option>
              <option
                value="Valladolid"
              >
                Valladolid
              </option>
              <option
                value="Vizcaya"
              >
                Vizcaya
              </option>
              <option
                value="Zamora"
              >
                Zamora
              </option>
              <option
                value="Zaragoza"
              >
                Zaragoza
              </option>
            </select>
          </div>
          <div>
            <label
              class="text-p-2 md:text-h-6 font-bold text-tile-100"
              for="numberOfPropertiesEnquire"
            >
              contact.form.properties_number.label
            </label>
            <select
              class="border-sierra-night-40 text-sierra-night-60 w-full rounded-md border p-2 pl-4 mt-2"
            >
              <option
                disabled=""
                selected=""
                value=""
              >
                contact.form.select.placeholder
              </option>
              <option
                value="5-50"
              >
                5-50
              </option>
              <option
                value="50-100"
              >
                50-100
              </option>
              <option
                value="100-200"
              >
                100-200
              </option>
              <option
                value="200-500"
              >
                200-500
              </option>
              <option
                value="500-1000"
              >
                500-1000
              </option>
            </select>
          </div>
        </div>
      </div>
      <div>
        <div
          class="mb-5 items-end lg:grid lg:grid-cols-2 lg:gap-x-6"
        >
          <div>
            <p
              class="text-p-2 md:text-h-6 mb-4 font-bold"
            >
              contact.form.arrange_callback
            </p>
            <div
              class="space-y-4 lg:flex lg:items-center lg:space-y-0 lg:space-x-4"
            >
              <label
                class="flex"
              >
                <input
                  class="relative border-sierra-night-40 relative inline-block h-6 w-6 appearance-none overflow-hidden rounded-full border-1 bg-white p-1 shadow-sm transition-all after:content[''] after:w-full after:h-full after:top-0 after:left-0 after:-right-0 after:bottom-0 after:m-auto after:rounded-full after:block after:bg-ocean-100 after:scale-0 after:transition-all checked:bg-white checked:after:scale-100"
                  name="timeOfDay"
                  type="radio"
                  value="Morning"
                />
                <span
                  class="text-p-2 ml-3"
                >
                  contact.form.date.morning
                </span>
              </label>
              <label
                class="flex"
              >
                <input
                  class="relative border-sierra-night-40 relative inline-block h-6 w-6 appearance-none overflow-hidden rounded-full border-1 bg-white p-1 shadow-sm transition-all after:content[''] after:w-full after:h-full after:top-0 after:left-0 after:-right-0 after:bottom-0 after:m-auto after:rounded-full after:block after:bg-ocean-100 after:scale-0 after:transition-all checked:bg-white checked:after:scale-100"
                  name="timeOfDay"
                  type="radio"
                  value="Afternoon"
                />
                <span
                  class="text-p-2 ml-3"
                >
                  contact.form.date.afternoon
                </span>
              </label>
            </div>
          </div>
          <div
            class="mt-5 lg:mt-0"
          >
            <p
              class="text-p-2 md:text-h-6 mb-4 font-bold"
            >
              contact.form.get_in_touch.label
            </p>
            <div
              class="space-y-4 lg:flex lg:items-center lg:space-y-0 lg:space-x-4"
            >
              <label
                class="flex"
              >
                <input
                  class="relative border-sierra-night-40 relative inline-block h-6 w-6 appearance-none overflow-hidden rounded-full border-1 bg-white p-1 shadow-sm transition-all after:content[''] after:w-full after:h-full after:top-0 after:left-0 after:-right-0 after:bottom-0 after:m-auto after:rounded-full after:block after:bg-ocean-100 after:scale-0 after:transition-all checked:bg-white checked:after:scale-100"
                  name="preferredFormOfContact"
                  type="radio"
                  value="Email"
                />
                <span
                  class="text-p-2 ml-3"
                >
                  contact.form.get_in_touch.email
                </span>
              </label>
              <label
                class="flex"
              >
                <input
                  class="relative border-sierra-night-40 relative inline-block h-6 w-6 appearance-none overflow-hidden rounded-full border-1 bg-white p-1 shadow-sm transition-all after:content[''] after:w-full after:h-full after:top-0 after:left-0 after:-right-0 after:bottom-0 after:m-auto after:rounded-full after:block after:bg-ocean-100 after:scale-0 after:transition-all checked:bg-white checked:after:scale-100"
                  name="preferredFormOfContact"
                  type="radio"
                  value="Phone call"
                />
                <span
                  class="text-p-2 ml-3"
                >
                  contact.form.get_in_touch.phone_call
                </span>
              </label>
            </div>
          </div>
        </div>
      </div>
      <input
        name="defaultNationContact"
        type="hidden"
        value="Spain"
      />
      <input
        name="hsLanguage"
        type="hidden"
        value="en"
      />
      <input
        name="receiveMarketAnalysis"
        type="hidden"
        value="true"
      />
      <footer
        class="text-h-5-sm lg:text-h-5 mt-3 w-full lg:w-1/4"
      >
        <button
          class="inline-block w-full rounded-full px-4 text-center font-bold text-white disabled:cursor-not-allowed bg-ocean-100 border-ocean-100 hover:border-ocean-120 hover:bg-ocean-120 focus:border-ocean-120 focus:bg-ocean-120 disabled:bg-ocean-40 disabled:border-ocean-40 border-2 py-3"
        >
          contact.form.submit
        </button>
      </footer>
    </form>
  </section>
</DocumentFragment>
`;
